devtools/x/src/build.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cargo::{CargoArgs, CargoCommand},
+    context::XContext,
+    utils,
+    utils::project_root,
+    Result,
+};
+use log::info;
+use std::{
+    ffi::OsString,
+    process::{Command, Stdio},
+};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(long, short, number_of_values = 1)]
+    /// Run test on the provided packages
+    package: Vec<String>,
+    #[structopt(long, short)]
+    /// Skip running expensive libra testsuite integration tests
+    unit: bool,
+    #[structopt(long)]
+    /// Test only this package's library unit tests, skipping doctests
+    lib: bool,
+    #[structopt(long)]
+    /// Do not fast fail the run if tests (or test executables) fail
+    no_fail_fast: bool,
+    #[structopt(long)]
+    /// Do not run tests, only compile the test executables
+    no_run: bool,
+    #[structopt(long, short)]
+    /// Number of parallel jobs, defaults to # of CPUs
+    jobs: Option<u16>,
+    #[structopt(long, parse(from_os_str))]
+    /// Directory to output HTML coverage report to (using grcov)
+    html_cov_dir: Option<OsString>,
+    #[structopt(name = "TESTNAME", parse(from_os_str))]
+    testname: Option<OsString>,
+    #[structopt(name = "ARGS", parse(from_os_str), last = true)]
+    args: Vec<OsString>,
+}
+
+pub fn run(mut args: Args, xctx: XContext) -> Result<()> {
+    args.args.extend(args.testname.clone());
+    let config = xctx.config();
+
+    let env_vars: &[(&str, &str)] = if args.html_cov_dir.is_some() {
```

Comment:
Does `xbuild` need to know anything about code coverage output? seems like that's specific to `xtest`

Reply:
grcov is text, yes, seemed reasonable to allow it all be installed up front to potentially replace parts of dev_setup.sh.